src/Netflix/Netflix.ts

```diff
@@ -0,0 +1,760 @@
+          }),
+        normal: false
+      },
+      {
+        name: "ChangeAudio",
+        match: ["[change/switch] audio to *", "audio to *"],
+        pageFn: (_: string, audioName?: string) =>
+          !!audioName &&
+          sendMessage({
+            key: "changeAudio",
+            sub: {
+              key: "ask",
+              query: audioName
+            }
+          }),
+        normal: false
+      },
+      {
+        name: "ChangeText",
+        match: ["[change/switch] [text/subtitle] to *", "[text/subtitle] to *"],
+        pageFn: (_: string, textName?: string) =>
+          !!textName &&
+          sendMessage({
+            key: "changeText",
+            sub: {
+              key: "ask",
+              query: textName
+            }
+          }),
+        normal: false
+      },
+      {
+        name: "Seek::To::Minute+Second",
+        match: ["skip to minute #", "skip to minute # second #"],
+        pageFn: (_: string, minute: number, second = 0) =>
+          sendMessage({
+            key: "skip",
+            sub: { key: "to", timestamp: (60 * minute + second) * 1000 }
+          }),
+        normal: false
+      },
+      {
+        name: "Seek::To::Second",
+        match: ["skip to second #"],
+        pageFn: (_: string, second: number) =>
+          sendMessage({
+            key: "skip",
+            sub: { key: "to", timestamp: second * 1000 }
+          }),
+        normal: false
+      },
+      {
+        name: "Seek::Ahead::Minute+Second",
+        match: [
+          "skip ahead # [minute/minutes]",
+          "skip ahead # [minute/minutes] # [second/seconds]"
+        ],
+        pageFn: (_: string, minute: number, second = 0) =>
+          sendMessage({
+            key: "skip",
+            sub: { key: "ahead", duration: (60 * minute + second) * 1000 }
+          }),
+        normal: false
+      },
+      {
+        name: "Seek::Ahead::Second",
+        match: ["skip ahead # [second/seconds]"],
+        pageFn: (_: string, second: number) =>
+          sendMessage({
+            key: "skip",
+            sub: { key: "ahead", duration: second * 1000 }
+          }),
+        normal: false
+      },
+      {
+        name: "Seek::Behind::Minute+Second",
+        match: [
+          "skip behind # [minute/minutes]",
+          "skip behind # [minute/minutes] # [second/seconds]"
+        ],
+        pageFn: (_: string, minute: number, second = 0) =>
+          sendMessage({
+            key: "skip",
+            sub: { key: "behind", duration: (60 * minute + second) * 1000 }
+          }),
+        normal: false
+      },
+      {
+        name: "Seek::Behind::Second",
+        match: ["skip behind # [second/seconds]"],
+        pageFn: (_: string, second: number) =>
+          sendMessage({
+            key: "skip",
+            sub: { key: "behind", duration: second * 1000 }
+          }),
+        normal: false
+      },
+
+      {
+        name: "Watch::Title",
+        match: ["watch *"],
```

Comment:
This is fine, but ideally we would only match valid shows. So you could cache the shows and then use a matchFn with the util's fuzzy matching algorithm to pick the right one. It might be performance prohibitive depending on how big the show db is.

Reply:
I am optimistic about devices that can run Chrome, can withhold fuzzy matching for tens of thousands less-than-100-bytes utf8 string. And shows stored on Netflix's cache is probably can reach just about hundreds.

Reply:
I also found out that LipSurf reads Aria-Label so I kind of utilize that rather than the fuzzy matching, but I guess I can pull out the data from Netflix cache back to the extension context ala "ChangeText" and "ChangeAudio".

Reply:
I ended up combining fuzzyHighScore with 0.8 minimum confidence and falling back to search if no matching videos. The overall UX feels better that way. I found a weird behavior with fuzzyHighScore though.

When I manually map an array of videos to fuzzyHighScore with the video as the only source and then sort the result in descending order, I got a different result to when I call the fuzzyHighScore with all videos put to sources.

```
const [index,title] = await fuzzyHighScores(query, videoTitles, 0, true);
```
returns a different result (highest score) than the code below
```
cosnt {index, title} = (await Promise.all(
  videoTitles.map(async title => ({
    score: await fuzzyHighScores(query, [title], 0, true)
  }))
))
  // Maps array index as the index, rather than the fuzzyIndex which is always 0
  .map(([_indexFromFuzzy, score], arrayIndex) => ({
    score,
    index: arrayIndex
  }))
  // Sort descending
  .sort(({scoreA},{scoreB}) => scoreB - scoreA)[0]
```

Reply:
"I ended up combining fuzzyHighScore with 0.8 minimum confidence and falling back to search if no matching videos. The overall UX feels better that way." that sounds good.
"I got a different result" -- that doesn't seem right. Can you give me the inputs you got those results with?

Reply:
Here you go.
There's also the code and result.
https://gist.github.com/Kelerchian/df50a53a9ef08cece06fdc326a9c1cfe